packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  IListenOptions,
+  LogLevelDesc,
+  Servers,
+} from "@hyperledger/cactus-common";
+import { SubstrateTestLedger } from "../../../../../cactus-test-tooling/src/main/typescript/substrate-test-ledger/substrate-test-ledger";
+import metadata from "../../rust/fixtures/ink/metadata.json";
+import fs from "fs";
+import { v4 as uuidv4 } from "uuid";
+import { pruneDockerAllIfGithubAction } from "@hyperledger/cactus-test-tooling";
+import express from "express";
+import http from "http";
+import {
+  PluginLedgerConnectorPolkadot,
+  IPluginLedgerConnectorPolkadotOptions,
+  DefaultApi as PolkadotApi,
+} from "../../../main/typescript/public-api";
+import { PluginKeychainMemory } from "@hyperledger/cactus-plugin-keychain-memory";
+import { PluginRegistry } from "@hyperledger/cactus-core";
+import { AddressInfo } from "net";
+import { Configuration } from "@hyperledger/cactus-core-api";
+import "jest-extended";
+
+const testCase = "invoke contract with all invocation types";
+describe(testCase, () => {
+  const logLevel: LogLevelDesc = "TRACE";
+
+  const DEFAULT_WSPROVIDER = "ws://127.0.0.1:9944";
+  const instanceId = "test-polkadot-connector";
+  const ledgerOptions = {
+    publishAllPorts: false,
+    logLevel: logLevel,
+    emitContainerLogs: true,
+  };
+  const ledger = new SubstrateTestLedger(ledgerOptions);
+  const expressApp = express();
+  expressApp.use(express.json());
+  expressApp.use(express.urlencoded({ extended: false }));
+  const server = http.createServer(expressApp);
+  let addressInfo: AddressInfo,
+    address: string,
+    port: number,
+    apiHost: string,
+    plugin: PluginLedgerConnectorPolkadot,
+    keychainEntryKey: string,
+    keychainEntryValue: string,
+    keychainPlugin: PluginKeychainMemory,
+    apiClient: PolkadotApi,
+    apiConfig: Configuration,
+    contractAddress: string;
+
+  const proofSize = 131072;
+  const refTime = 6219235328;
+  const gasLimit = {
+    refTime,
+    proofSize,
+  };
+  beforeAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).toResolve();
+  });
+  afterAll(async () => {
+    await ledger.stop();
+    await plugin.shutdownConnectionToSubstrate();
+  });
+  afterAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).resolves.toBeTruthy();
+  });
+  afterAll(async () => await Servers.shutdown(server));
+  beforeAll(async () => {
+    await ledger.start();
+    expect(ledger).toBeTruthy();
```

Comment:
@AnmolBansalDEV It would be more useful to check for the output of the `ledger.start()` to be checked for truthyness because the program will 100% crash before this assertion if the `ledger` object was not truthy (because we already tried to call a method on it in the previous line)

Reply:
updated
https://github.com/hyperledger/cacti/blob/87ec994040fade0e7a4a690883c971baa9eb9702/packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts#L74-L75